### ISSUE TITLE
Incorrect url when loading components from different apps

### ISSUE THREAD

**Issue by Krzyrok** [ionic-team/stencil#2246]
**Stencil version:**

```
 @stencil/core@1.9.0-21 and regular @1.8.5
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**
We have library with web-components which is built by stencil. We use these components in our react apps. Let's say we have App1 and App2 - both built with react stack. Both are hosted independently: `http://address-to-APP1` and `http://address-to-APP2`. But now we want to share some react components between apps without moving them e.g. we want to use `App2.ComponentB` in `App1`. We have such functionality - we load component from App2 in App1. It works correctly for plain react components (webpack handles loading chunks without troubles). But problem occurs if we use some web-component in `App2.ComponentB`. It should be loaded from `http://address-to-APP2`. But it is incorrectly loaded from APP1 address :/

I tried to use `resourcesUrl` in `defineCustomElements` but it does not have any effect.  

~~Stencil loader looks for first/last script on the page (as I observed) and uses URL from it to load web-components. Generated URLs (src of script tag) for lazy loaded web-components have `http://address-to-APP1` or `/` base url (it's based on other scripts generated from webpack as I mentioned).~~ UPDATE: https://github.com/ionic-team/stencil/issues/2246#issuecomment-597530711

**Expected behavior:**
URL for loading web-components (built by stencil) should be configurable in the JS.  
web-components loaded through js which is generated by App1 should be taken from `http://address-to-APP1`.
web-components loaded through js which is generated by App2 should be taken from `http://address-to-APP2` - even if App2.js was loaded and executed by App1.  

Flow:
http://address-to-APP1 ->(loads) App1 -> ComponentA -> web-component: fetched from http://address-to-APP1 - works OK

http://address-to-APP2 ->(loads) App2 -> ComponentB -> web-component: fetched from http://address-to-APP2 - works OK

http://address-to-APP1 ->(loads) App1 -> ComponentA -> (loads) App2 (from http://address-to-APP2) -> ComponentB -> web-component: fetched from http://address-to-APP1 - works INCORRECT - web-component should loaded from http://address-to-APP2 (or other configurable address)

**Steps to reproduce:**
1. create components library with stencil usage (dist output target)
2. import these components in the App1
```tsx
import { defineCustomElements } from '<your-library>/dist/loader';
defineCustomElements(window, { resourcesUrl: 'http://address-to-APP1' });
```
3. import also these components in the App2 (App1 and App2 are independent apps - they can use different versions of components library from the above).
```tsx
import { defineCustomElements } from '<your-library>/dist/loader';
defineCustomElements(window, { resourcesUrl: 'http://address-to-APP2' });
```
4. in the App1 load some component from App2 which uses web-components.

**Other information:**
I was expected that `resourcesUrl` allows to specify URL where web-components are hosted. If this is true, then it does not work.
Here is another bug for `resourcesUrl`: https://github.com/ionic-team/stencil/issues/1846  

If there is another approach to achieve the goal - let me know.
  
Looks also similar to:
https://github.com/ionic-team/stencil/issues/793
https://github.com/ionic-team/stencil/issues/1456

**Comment by Krzyrok** [comment]
In the linked bugs, there were suggestions that:
1) last script on the page should have correct URL (the same where web-component is located). But I don't have control on the scripts on the page. I don't add them manually or even dynamically. Some scripts are added by webpack loader (React environment) when e.g. new chunk should be loaded.
2) Here is a suggestion https://github.com/ionic-team/stencil/pull/1788#issuecomment-518589428 about adding script tag with `data-stencil-namespace` attribute. But as I mentioned - I do not add scripts to the page. I use regular ES modules and `import` syntax - all translated by webpack. The best would be the option to configure URL through JS - not by inserting additional empty script tag.

**Comment by Krzyrok** [comment]
What I've discovered:  
1. loading chunks with web-components is done through through `loadModule` function https://github.com/ionic-team/stencil/blob/master/src/client/client-load-module.ts#L21
2. dynamic import is handled by webpack.
3. To have correct URLs in one project (e.g. App1 -> ComponentA -> web-component) you can use webpack's `output.publicPath`: https://webpack.js.org/configuration/output/#outputpublicpath
  
  
But problem occurs if you try to load ComponentA (App1) and ComponentB (App2). Both in App1. Some of the things are loaded correctly. But when rendering ComponentB, funciton from react-dom is fired:
```
function appendChild(parentInstance, child) {
  parentInstance.appendChild(child);
}
```
It is from `http://address-to-APP2` (APP2: correct).  
But next is `HostElement.prototype.connectedCallback` from stencil. And it is run from `http://address-to-APP1` (APP1). At the end, there is stencil's `loadModule` function (mentioned above) with webpack's dynamic import - which is also from `http://address-to-APP1` (APP1). So it tries to load chunks from `http://address-to-APP1` (APP1: incorrect). And this causes problems.

**Comment by Krzyrok** [comment]
OK - we've got some agreement in the team about incorrect assumptions:  current definition of web components assume global registry for web components. If we want to load the same library with web-components multiple times in different versions then we need some mechanism of scoping web-components.  
  
There is a proposition for scoped web-components (scoped registry): https://github.com/w3c/webcomponents/issues/716  
But this is a proposition for now.  
  
There are also some alternative libraries which allows for scoping web-components: https://open-wc.org/scoped-elements/  
But this connected with different library for web-components (lit-html and lit-element) - so changing whole library is a big change.

Another, first satisfying, option is to modify every custom element during build and add some prefix specific to usage. In our case it would be e.g. `app1` in APP1 and `app2` in APP2. It would require changes of every usage (modification in build pipeline for TSX files) and modification of custom elements definition (in some e.g. postinstall script).  
  
But we picked another option: use just one version of library with web-components which is provided by main app, e.g APP1 which loads APP2. APP2 calls `defineCustomElements` only if it is run in "standalone" version. If this is loaded to APP1, then APP1 calls `defineCustomElements`. This require to use the same version of library in all apps, of course.

So I'm closing this issue. But still `resourcesUrl` is mysterious for me: it doesn't matter what I set as this parameter - it doesn't affect URLs (maybe the reason is that I set webpack's `output.publicPath`).
And I'm a little bit sad that nobody responded after almost 2 weeks :/
